### PR TITLE
Update scenarios documentation for .NET 9

### PR DESCRIPTION
**Pull request by lbussell**
Related: #5978

### CODE REVIEW

File: documentation/scenarios/installing-dotnet.md [hunk 1]
```diff
@@ -2,65 +2,80 @@
...
+In some cases, you can work around the need for the .NET runtime by publishing your application as a [self-contained app](https://learn.microsoft.com/en-us/dotnet/core/deploying/), in which case all of your app's dependencies are packaged with the app. This reduces the dependencies that need to be installed separately on the base image. Example Dockerfiles that demonstrate publishing a self-contained app are available in the [releasesapp sample](https://github.com/dotnet/dotnet-docker/blob/main/samples/releasesapp/README.md).
```
Comment (mthalman) [inline documentation/scenarios/installing-dotnet.md:13]:
```suggestion
In some cases, you can work around the need for the .NET runtime by publishing your application as a [self-contained app](https://learn.microsoft.com/en-us/dotnet/core/deploying/), in which case all of your app's dependencies are packaged with the app. This reduces the dependencies that need to be installed separately on the base image. Example Dockerfiles that demonstrate publishing a self-contained app are available in the [releasesapp sample](../../samples/releasesapp/README.md).
```

File: documentation/scenarios/installing-dotnet.md [hunk 2]
```diff
@@ -95,39 +115,67 @@ This provides full transparency to consumers of the image in regard to where the
...
+However, you should only use one or the other to install .NET.
+
+Starting with .NET 9, we have stopped publishing .NET packages on `packages.microsoft.com` for distros that publish their own .NET packages.
+For more details please see [.NET 9 Package Publishing announcement](https://github.com/dotnet/core/discussions/9556).
+
+.NET is supported by Microsoft when downloaded from a Microsoft source. Best effort support is offered from Microsoft when downloaded from elsewhere. You can open issues at [dotnet/core](https://github.com/dotnet/core/issues/new) if you run into problems.
+
+#### Native Package Installation Example
+
+Note: distro-archive packages are typically limited to the `.1xx` SDK band, while packages.microsoft.com packages are not. Users that want to use the latest feature band builds (on distros without packages.microsoft.com packages available) must install those builds manually via [binary archive](#installing-from-a-binary-archive) or [dotnet-install script](#installing-from-dotnet-install-script).
```
Comment (mthalman) [inline documentation/scenarios/installing-dotnet.md:144]: This should be placed after the Dockerfile example below. Having it above makes it seem like the Dockerfile example is illustrating the installation of a non-.1xx band.

File: documentation/scenarios/installing-dotnet.md
Comment (mthalman) [inline documentation/scenarios/installing-dotnet.md:1]: Please remove all usage of `en-us` in the URLs.

File: documentation/scenarios/installing-dotnet.md [hunk 3]
```diff
@@ -95,39 +115,67 @@ This provides full transparency to consumers of the image in regard to where the
...
+
+Keep in mind the [Dockerfile best practices](https://docs.docker.com/build/building/best-practices/) when porting the installation commands to your Dockerfile - in particular, installing or using `sudo` in your Dockerfile is not neccesary since the default user in most base container images is `root`.
+
+Here is an example Dockerfile using Debian:
 
 ```Dockerfile
-FROM ubuntu:focal
-RUN export DEBIAN_FRONTEND=noninteractive \
-    apt-get update \
+FROM amd64/debian:bookworm-slim
```
Comment (mthalman) [inline documentation/scenarios/installing-dotnet.md:163]: Is there a reason this needs to be arch-specific?

```suggestion
FROM debian:bookworm-slim
```

File: documentation/scenarios/installing-dotnet.md [hunk 4]
```diff
@@ -95,39 +115,67 @@ This provides full transparency to consumers of the image in regard to where the
...
+RUN apt-get update \
     # Install prerequisites
     && apt-get install -y --no-install-recommends \
        wget \
        ca-certificates \
     \
     # Install Microsoft package feed
-    && wget -q https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb \
+    wget -q https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb -O packages-microsoft-prod.deb \
```
Comment (mthalman) [inline documentation/scenarios/installing-dotnet.md:171]:
```suggestion
    && wget -q https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb -O packages-microsoft-prod.deb \
```

File: documentation/scenarios/installing-dotnet.md (hunk 3 above)
Comment (lbussell) [reply documentation/scenarios/installing-dotnet.md:163]: No, I just copied it from a different non-Ubuntu Dockerfile.

**Review by mthalman — APPROVED** [review]

**Pull request merged**